src/main/java/entity/PostFactory.java

```diff
@@ -0,0 +1,22 @@
+package entity;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PostFactory {
+    public Post createPost(User author, String content, String attachmentPath, String fileType, 
+                           String postTitle, String category) {
+        Content postContent = new PostContent(content, attachmentPath, fileType);
+        String entryID = generateID();
+        List<Comment> comments = new ArrayList<Comment>();
+        Post post = new Post(entryID, author, postContent, LocalDateTime.now(), null, 
+                             0, 0, postTitle, comments, category);
+        return post;
+    }
+
+    private String generateID() {
+        // TODO generate ID using java.util.UUID and check if it exists in the database?
```

Comment:
I see that [UUID generates](https://docs.oracle.com/javase/8/docs/api/index.html?java/util/UUID.html) a UUID class... do we want to change the types of all IDs to type java.util.UUID?

Reply:
There should be a toString method for it. I'm just keeping it here for reference, because I'm not sure about a good way to detect id collisions. We might end up generating it in the controllers and pass it in, or just check from the database for collisions.

Reply:
Oh yup, I just mean if we want to compare betwen UUIDs, it might be easier to keep things as type UUID instead of doing string matching right? And there might be additional functionality in type UUID that might be useful to keep as we write the rest of our code

Reply:
Yea it could, but we probably want to store the id in the database as well, so having a string would be easier? We should worry less about our program efficiency, but more about having something that works.

Reply:
Oh yup yup - we can always just cast to a string only at the point when we want to save it in a DB. 

I guess if we don't expect a lot of handling of UUIDs casting to string right away is good too!